File: scratch/process_logo.js
const Jimp = require('jimp');

async function processLogo() {
  try {
    const image = await Jimp.read('public/logo-wide.png');
    
    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
      const r = this.bitmap.data[idx + 0];
      const g = this.bitmap.data[idx + 1];
      const b = this.bitmap.data[idx + 2];
      const a = this.bitmap.data[idx + 3];

      // If mostly white background -> make transparent
      if (r > 240 && g > 240 && b > 240) {
        this.bitmap.data[idx + 3] = 0; // Alpha to 0
      } 
      // If mostly dark/black (text and dark parts) -> make white
      else if (r < 80 && g < 100 && b < 100) {
        this.bitmap.data[idx + 0] = 255;
        this.bitmap.data[idx + 1] = 255;
        this.bitmap.data[idx + 2] = 255;
      }
      // Everything else (like the green) stays exactly the same
    });

    await image.writeAsync('public/logo-wide-processed.png');
    console.log('Successfully processed logo to public/logo-wide-processed.png');
  } catch (error) {
    console.error('Error processing image:', error);
  }
}

processLogo();

File: scratch/process_logo.py
from PIL import Image
import sys

def process_logo(input_path, output_path):
    img = Image.open(input_path).convert("RGBA")
    datas = img.getdata()

    newData = []
    for item in datas:
        # item is (R, G, B, A)
        r, g, b, a = item
        
        # If the pixel is mostly white (background), make it transparent
        if r > 240 and g > 240 and b > 240:
            newData.append((255, 255, 255, 0))
        # If the pixel is dark (the text and dark parts of the logo), make it white
        elif r < 80 and g < 100 and b < 100:
            # We keep the original alpha just in case, but usually it's 255
            # We want to make it white
            newData.append((255, 255, 255, a))
        else:
            # Leave the green pixels (and any other colored pixels) mostly intact
            newData.append(item)

    img.putdata(newData)
    img.save(output_path, "PNG")
    print(f"Successfully processed logo and saved to {output_path}")

if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("Usage: python process_logo.py <input> <output>")
    else:
        process_logo(sys.argv[1], sys.argv[2])
